src/game.py

```diff
@@ -0,0 +1,301 @@
+from threading import Thread, Event
+
+from src.game_map.map import Map
+from src.players.player import Player
+from src.players.player_manager import PlayerManager
+from src.remote.game_client import GameClient
+
+
+class Game(Thread):
+    def __init__(self, game_name: str | None = None, num_turns: int | None = None,
+                 max_players: int = 1, is_full: bool = False) -> None:
+        super().__init__()
+
+        self.game_map: Map | None = None
+        self.__game_name: str | None = game_name
+
+        # create an active event
+        self.over: Event = Event()
+
+        self.__num_turns: int | None = num_turns
+        self.__max_players: int = max_players
+        self.__is_full: bool = is_full
+        self.__num_rounds: int | None = None
+        self.__current_round: int | None = None
+        self.__next_round: bool = True
+
+        self.__winner: int | None = None
+        self.__winner_index: int | None = None
+        self.__started: bool = False
+        self.__game_is_draw: bool = False
+
+        self.__connection_error: ConnectionError | None = None
+
+        self.__current_turn: list[int] = [-1]
```

Comment:
better use Dict/List from typing

Reply:
What exactly is the rationale behind this?
I looked into it, and by PEP 585: https://peps.python.org/pep-0585/, which is accepted for Python>= 3.9, it says that importing these from typing is deprecated:
![slika](https://user-images.githubusercontent.com/81580576/236775895-67455378-6ead-4483-9e09-dedd2e8fde95.png)
Again, if necessary I have no issues with changing these types to the ones from typing, but I was just going by the official Python guidelines.